nexus/Freqlog/Freqlog.py

```diff
@@ -122,27 +130,26 @@ def _log_and_reset_word(min_length: int = 2) -> None:
         while self.is_logging:
             try:
                 action: ActionType
-                key: kbd.Key | kbd.KeyCode | mouse.Button
+                modifiers: vinput.KeyboardModifiers
                 time_pressed: datetime
 
                 # Blocking here makes the while-True non-blocking
-                action, key, time_pressed = _get_timed_interruptable(self.q, self.new_word_threshold)
+                action, key, modifiers, time_pressed = _get_timed_interruptable(self.q, self.new_word_threshold)
+
+                if isinstance(key, bytes):
+                    key = key.decode('utf-8')
 
                 # Debug keystrokes
-                if isinstance(key, kbd.Key) or isinstance(key, kbd.KeyCode):
+                if key != '' and key != '\0':
                     logging.debug(f"{action}: {key} - {time_pressed}")
-                    logging.debug(f"word: '{word}', active_modifier_keys: {active_modifier_keys}")
+                    logging.debug(f"word: '{word}', modifiers: {modifiers}")
 
-                # Update modifier keys
-                if action == ActionType.PRESS and key in self.modifier_keys:
-                    active_modifier_keys.add(key)
-                elif action == ActionType.RELEASE:
-                    active_modifier_keys.discard(key)
+                if action == ActionType.RELEASE:
+                    continue
```

Comment:
There are 3+ occurrences of `key != '' and key != '\0'` checks. In what situations will key be those values, and can we just `continue` here if `key in {'', '\0'}`?